src/deq_eid/main.py

```diff
@@ -0,0 +1,363 @@
+        #: (all log messages were duplicated if put at beginning)
+        logging.captureWarnings(True)
+
+        skid_logger.debug("Creating Supervisor object")
+        self.supervisor = Supervisor(handle_errors=False)
+        sendgrid_settings = config.SENDGRID_SETTINGS
+        sendgrid_settings["api_key"] = self.secrets.SENDGRID_API_KEY
+        self.supervisor.add_message_handler(
+            SendGridHandler(
+                sendgrid_settings=sendgrid_settings, client_name=config.SKID_NAME, client_version=version.__version__
+            )
+        )
+
+    def _remove_log_file_handlers(self):
+        """A helper function to remove the file handlers so the tempdir will close correctly"""
+
+        loggers = [logging.getLogger(config.SKID_NAME), logging.getLogger("palletjack")]
+
+        for logger in loggers:
+            for handler in logger.handlers:
+                try:
+                    if self.log_name in handler.stream.name:
+                        logger.removeHandler(handler)
+                        handler.close()
+                except Exception:
+                    pass
+
+    def _get_incidents(self) -> GeoAccessor:
+        self.skid_logger.info("loading environmental incident records from Salesforce...")
+        records = helpers.SalesForceRecords(
+            self.salesforce_extractor,
+            config.INCIDENTS_SF_API,
+            config.INCIDENTS_FIELDS,
+            where_clause="Utm_N_Y_7_dgts__c != null and Utm_E_X_6_dgts__c != null",
+        )
+        records.extract_data_from_salesforce()
+
+        self.skid_logger.info("converting to spatial dataframe...")
+        sdf = GeoAccessor.from_xy(records.df, "Easting", "Northing", sr=26912)
+
+        self.skid_logger.info("projecting...")
+        sdf.spatial.project(3857, "NAD_1983_To_WGS_1984_5")
+        sdf.spatial.sr = arcgis.geometry.SpatialReference(wkid=3857)
+
+        sdf["Date_Discovered"] = sdf["Date_Discovered_For_Filter"].dt.strftime("%m/%d/%Y")
+
+        sdf = sdf.query("Northing > 0 & Easting > 0")
+
+        return sdf
+
+    def _get_chemical(self) -> GeoAccessor:
+        self.skid_logger.info("loading chemicals records from Salesforce...")
+        records = helpers.SalesForceRecords(
+            self.salesforce_extractor,
+            config.CHEMICAL_SF_API,
+            config.CHEMICAL_FIELDS,
+            None,
+        )
+        records.extract_data_from_salesforce()
+
+        return records.df
+
+    def _get_media(self) -> GeoAccessor:
+        self.skid_logger.info("loading impacted media records from Salesforce...")
+        records = helpers.SalesForceRecords(
+            self.salesforce_extractor,
+            config.MEDIA_SF_API,
+            config.MEDIA_FIELDS,
+            None,
+        )
+        records.extract_data_from_salesforce()
+
+        return records.df
+
+    def _publish_dataset(self, table_name, title, fields, sdf, type):
```

Comment:
question: does this method get called in cloud run? If so, how are you getting arcpy? If you need to write to GDBs, you can crib code from palletjack, which uses geopandas for access and is thus pip-installable in GCP.

Reply:
No, it does not get called in cloud run.